ui/e2e/instances/details/index.spec.ts

```diff
@@ -250,3 +259,194 @@ test("the diagram on the instance page changes appearance dynamically", async ({
+  if (minimizedWidth === undefined || maximizedWidth === undefined) {
+    throw new Error("could not get width of diagram panel");
+  }
+  expect(
+    maximizedWidth / minimizedWidth,
+    "The panel is significantly bigger after maximizing"
+  ).toBeGreaterThan(1.5);
+
+  await resizeButton.hover();
+  await expect(
+    page.getByText("minimize output"),
+    "it shows the correct text when hovering over the resize button"
+  ).toBeVisible();
+
+  await page.reload();
+
+  const currentWidthAfterReload = (await inputOutputPanel.boundingBox())?.width;
+  expect(
+    currentWidthAfterReload,
+    "after reloading the page, the panel is still maximized"
+  ).toEqual(maximizedWidth);
+
+  await resizeButton.click();
+  await inputButton.click();
+
+  await expect(textarea, "the text shows the expected input").toHaveText(
+    expectedInput
+  );
+
+  await outputButton.click();
+
+  await expect(textarea, "the text shows the expected output").toHaveText(
+    expectedOutput
+  );
+
+  // for writing to the clipboard and for reading from the clipboard we need to grant permissions here - if we are not allowing it universally in the playwright.config.ts
+  page.context().grantPermissions(["clipboard-write", "clipboard-read"]);
+
+  await copyButton.click();
+
+  expect(await page.evaluate(() => navigator.clipboard.readText())).toEqual(
+    expectedOutputCopy
+  );
+
+  await inputButton.click();
+  await copyButton.click();
+
+  expect(await page.evaluate(() => navigator.clipboard.readText())).toEqual(
+    expectedInput
+  );
+});
+
+test("the output shows event notifications", async ({ page }) => {
+  const newInstance = createInstance({
+    namespace,
+    path: failingWorkflowName,
+  });
+
+  await expect(newInstance, "wait until process was completed").toBeDefined();
+  const instanceId = (await newInstance).instance;
+  await page.goto(`/${namespace}/instances/${instanceId}`);
+
+  const inputOutputPanel = page.getByTestId("inputOutputPanel");
+
+  await expect(inputOutputPanel).toBeVisible();
+
+  const outputButton = inputOutputPanel
+    .getByRole("tablist")
+    .locator("button")
+    .nth(1);
+
+  const header = page.getByTestId("instance-header-container");
+  const textarea = inputOutputPanel.locator(".view-lines");
+
+  const runningInstanceOutput = "The workflow is still running";
+  const failedInstanceOutput = '{    "result": "an error occurred"}';
+
+  await outputButton.click();
+
+  await expect(
+    inputOutputPanel,
+    "the output shows the text for the status running instance"
+  ).toContainText(runningInstanceOutput);
+
+  await expect(
+    header.locator("div").first(),
+    "the badge failed is visible"
+  ).toContainText("failed");
+
+  await expect(
+    textarea,
+    "the output shows the text for the status failed instance"
+  ).toHaveText(failedInstanceOutput);
+});
+
+test("after a running instance finishes, the output tab is automatically selected ", async ({
```

Comment:
This test fails for me all the time. It looks like the instance is always already finished when the page loads. This might be depending on the performance of the machine that runs the tests. Changing the workflow to use `duration: PT8S` works for me. Not 100 % sure how we deal with this. Having long-running workflows in our test suite will add up to a very long times for the tests to run.

![CleanShot 2024-04-18 at 09 54 36@2x](https://github.com/direktiv/direktiv/assets/121789579/6e854903-bff2-4b02-be85-d15a1827ec8a)

Reply:
I guess we need to update this to PT8S before we merge, still can not get this test to pass, and I don't have a better idea.